### ISSUE TITLE
deletion attempts cause deadlock

### ISSUE THREAD

**Issue by lukemarsden** [dotmesh-io/dotmesh#358]
If a filesystem is marked for deletion and deletion occurs in `initFilesystemMachine`, the evils of mutexes mean that we attempt to acquire twice.

This is affecting our demos, and could well affect users too.

**Comment by lukemarsden** [comment]
After fixing the deadlock, the following logs emerged:

```
I  2018/03/27 17:12:56 [initFilesystemMachine] starting: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [initFilesystemMachine] acquired lock: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Attempting to delete filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Errors deleting filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4, possibly because some operations were previously completed: [error reading error: [99 97 110 110 111 116 32 111 112 101 110 32 39 112 111 111 108 47 100 109 102 115 47 54 54 97 98 102 54 97 55 45 49 100 98 98 45 52 54 52 57 45 52 99 97 98 45 97 55 100 51 54 53 56 56 54 56 98 52 39 58 32 100 97 116 97 115 101 116 32 100 111 101 115 32 110 111 116 32 101 120 105 115 116 10]]
 
I  2018/03/27 17:12:56 UNMOUNTING: 66abf6a7-1dbb-4649-4cab-a7d3658868b4=6db4d3d65085648e
 
I  2018/03/27 17:12:56 Error in runForever(fetchAndWatchEtcd), retrying in 1s: No such filesystemId 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [initFilesystemMachine] starting: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [initFilesystemMachine] acquired lock: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Attempting to delete filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Errors deleting filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4, possibly because some operations were previously completed: [error reading error: [99 97 110 110 111 116 32 111 112 101 110 32 39 112 111 111 108 47 100 109 102 115 47 54 54 97 98 102 54 97 55 45 49 100 98 98 45 52 54 52 57 45 52 99 97 98 45 97 55 100 51 54 53 56 56 54 56 98 52 39 58 32 100 97 116 97 115 101 116 32 100 111 101 115 32 110 111 116 32 101 120 105 115 116 10]]
 
I  2018/03/27 17:12:56 UNMOUNTING: 66abf6a7-1dbb-4649-4cab-a7d3658868b4=6db4d3d65085648e
 
I  2018/03/27 17:12:56 Error in runForever(fetchAndWatchEtcd), retrying in 1s: No such filesystemId 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [initFilesystemMachine] starting: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [initFilesystemMachine] acquired lock: 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Attempting to delete filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4
 
I  2018/03/27 17:12:56 [deleteFilesystem] Errors deleting filesystem 66abf6a7-1dbb-4649-4cab-a7d3658868b4, possibly because some operations were previously completed: [error reading error: [99 97 110 110 111 116 32 111 112 101 110 32 39 112 111 111 108 47 100 109 102 115 47 54 54 97 98 102 54 97 55 45 49 100 98 98 45 52 54 52 57 45 52 99 97 98 45 97 55 100 51 54 53 56 56 54 56 98 52 39 58 32 100 97 116 97 115 101 116 32 100 111 101 115 32 110 111 116 32 101 120 105 115 116 10]]
 
I  2018/03/27 17:12:56 MOUNTING: 66abf6a7-1dbb-4649-4cab-a7d3658868b4=6db4d3d65085648e
 
I  2018/03/27 17:12:56 Error in runForever(fetchAndWatchEtcd), retrying in 1s: No such filesystemId 66abf6a7-1dbb-4649-4cab-a7d3658868b4
```

https://github.com/dotmesh-io/dotmesh/commit/fa1cf3ec7b625095e60a33c9f68965fe060a7119 fixes the strings-of-ints, and the fix for the infinite loop is about to land...

**Comment by lukemarsden** [comment]
there it is!

**Comment by prisamuel** [comment]
Tests for edge cases will be addressed in #360
